src/util-decode-mime.c

```diff
@@ -2269,17 +2264,16 @@ static int ProcessMimeBody(const uint8_t *buf, uint32_t len,
     int body_found = 0;
     uint32_t tlen;
 
-#ifdef HAVE_NSS
-    if (MimeDecGetConfig()->body_md5) {
-        if (state->body_begin == 1) {
-            if (state->md5_ctx == NULL) {
-                state->md5_ctx = HASH_Create(HASH_AlgMD5);
-                HASH_Begin(state->md5_ctx);
+    if (!g_disable_hashing) {
+        if (MimeDecGetConfig()->body_md5) {
+            if (state->body_begin == 1) {
+                if (state->md5_ctx == NULL) {
+                    state->md5_ctx = SCMd5New();
```

Comment:
can this fail?

Reply:
No, not in a way we can test anyways.  Much like any pointer that is created in Rust and passed back to see.

The same actually applies to the jsonbuilder stuff.  Just looking now, we do check for null with the jsonbuilder stuff, but AFAIK, it will never actually return a NULL, as the Box::into_raw we use never will return NULL.  We could add the NULL check, but its just non-functional sugar, but I admit, it does look better.